framework/include/ics/InitialConditionBase.h

```diff
@@ -91,13 +91,18 @@ class InitialConditionBase : public MooseObject,
 
   virtual const std::set<std::string> & getSuppliedItems() override;
 
+  int & getMyState();
```

Comment:
I tried this, but when looping through the states, it leaves more room for errors. When the unsigned int is reduced each time during the for loops, eventually it wants to become negative, indicating the end of the loop has been reached. But because it is unsigned, the remains positive and causes an endless loop. There are solutions for this of course, but it will create room for mistakes if someone wants to change the logic. I think we should leave all of them as signed ints.

Reply:
I'll implement it this way. But I respectfully disagree. Casting to signed ints explicitly just for the comparison operators, or adding 1 and ending the loop at 1 are the two simple ways of fixing the problem. And both seem less perfect than having signed ints. Especially when someone might make the mistake again in the future. We could even save more storage space by making them shorts or bytes in if the data is what you're concerned about.

Reply:
Hopefully my concerns won't be important anymore though when I implement your new logic for applying old and older ICs.